pkg/filestore/minio/minio.go

```diff
@@ -100,7 +101,15 @@ func (s *Store) EnsureBucket(ctx context.Context) error {
 }
 
 func (s *Store) NewReader(ctx context.Context, path string) (rc io.ReadCloser, err error) {
-	// No error returned even if the object does not exist.
+	_, err = s.client.StatObject(ctx, s.bucket, path, minio.GetObjectOptions{})
+	if err != nil {
+		errRes := minio.ToErrorResponse(err)
+		if errRes.StatusCode == http.StatusNotFound {
+			err = filestore.ErrNotFound
+		}
+		s.logger.Error("failed to create minio object reader", zap.String("path", path), zap.Error(err))
```

Comment:
nit: It would be great if it could be followed our style where must be indented if multiple args given:

```go
	s.logger.Error("failed to create minio object reader",
		zap.String("path", path),
		zap.Error(err),
	)
```